pkg/cli/server/root.go

```diff
@@ -861,8 +861,8 @@ func readLDAPCredentials(ldapConfigPath string) (config.LDAPCredentials, error)
 
 	var ldapCredentials config.LDAPCredentials
 
-	if err := viperInstance.Unmarshal(&ldapCredentials); err != nil {
-		log.Error().Err(err).Msg("failed to unmarshal new config")
+	if err := viperInstance.UnmarshalExact(&ldapCredentials); err != nil {
+		log.Error().Err(err).Msg("failed to unmarshal ldap credentials config")
```

Comment:
Please add a test to cover this line. It should be easier than before, since we should fail in case of invalid or missing fields.

Reply:
Added new tests to ensure that credentials file contains all required fields or has any extra fields.

Please note that it does not check if [credentialsFile](https://github.com/project-zot/zot/pull/2215/files#diff-464834167072cf556f61bd7ba8a129ccde65577dc94ac2b0ce0ddecfb21d864cR21) field is present in `auth` config. Let me know if we want to change this old behavior and make it required.